// @kamiazya/ngx-speech-synthesis/esm2015/lib/services/speech-synthesis-utterance-factory.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable, Optional, Inject } from '@angular/core';
import { Lang, Voice, Volume, Rate, Pitch, OnStartHandler, OnEndHandler, OnErrorHandler, OnPauseHandler, OnResumeHandler, OnMarkHandler, OnBoundaryHandler, Config, } from '../configs';
import * as i0 from "@angular/core";
import * as i1 from "../configs";
/**
 * \@dynamic
 */
export class SpeechSynthesisUtteranceFactoryService {
    /**
     * @param {?=} config
     * @param {?=} lang
     * @param {?=} voice
     * @param {?=} volume
     * @param {?=} rate
     * @param {?=} pitch
     * @param {?=} onstart
     * @param {?=} onend
     * @param {?=} onerror
     * @param {?=} onpause
     * @param {?=} onresume
     * @param {?=} onmark
     * @param {?=} onboundary
     */
    constructor(config, 
    /**
     * This attribute specifies the language of the speech synthesis for the utterance,
     * using a valid BCP 47 language tag.
     * [BCP47] If unset it remains unset for getting in script,
     * but will default to use the language of the html document root element and associated hierarchy.
     * This default value is computed and used when the input request opens a connection
     * to the recognition service.
     */
    lang, 
    /**
     * This attribute specifies the speech synthesis voice that the web application wishes to use.
     * When a SpeechSynthesisUtterance object is created this attribute must be initialized to null.
     * If, at the time of the speak() method call,
     * this attribute has been set to one of the SpeechSynthesisVoice objects returned by getVoices(),
     * then the user agent must use that voice. If this attribute is unset or null at the time of the speak()
     * method call, then the user agent must use a user agent default voice.
     * The user agent default voice should support the current language (see lang) and
     * can be a local or remote speech service and can incorporate end user choices via interfaces
     * provided by the user agent such as browser configuration parameters.
     */
    voice, 
    /**
     * This attribute specifies the speaking volume for the utterance.
     * It ranges between 0 and 1 inclusive, with 0 being the lowest volume and 1 the highest volume,
     * with a default of 1. If SSML is used, this value will be overridden by prosody tags in the markup.
     */
    volume, 
    /**
     * This attribute specifies the speaking rate for the utterance.
     * It is relative to the default rate for this voice.
     * 1 is the default rate supported by the speech synthesis engine or specific voice
     * (which should correspond to a normal speaking rate).
     * 2 is twice as fast, and 0.5 is half as fast. Values below 0.1 or above 10 are strictly disallowed,
     * but speech synthesis engines or specific voices may constrain the minimum and maximum rates further,
     * for example, a particular voice may not actually speak faster than 3 times normal
     * even if you specify a value larger than 3.
     * If SSML is used, this value will be overridden by prosody tags in the markup.
     */
    rate, 
    /**
     * This attribute specifies the speaking pitch for the utterance.
     * It ranges between 0 and 2 inclusive, with 0 being the lowest pitch and 2 the highest pitch.
     * 1 corresponds to the default pitch of the speech synthesis engine or specific voice.
     * Speech synthesis engines or voices may constrain the minimum and maximum rates further.
     * If SSML is used, this value will be overridden by prosody tags in the markup.
     */
    pitch, 
    /**
     * Fired when this utterance has begun to be spoken.
     */
    onstart, 
    /**
     * Fired when this utterance has completed being spoken. If this event fires,
     * the error event must not be fired for this utterance.
     */
    onend, 
    /**
     * Fired if there was an error that prevented successful speaking of this utterance.
     * If this event fires, the end event must not be fired for this utterance.
     */
    onerror, 
    /**
     * Fired when and if this utterance is paused mid-utterance.
     */
    onpause, 
    /**
     * Fired when and if this utterance is resumed after being paused mid-utterance.
     * Adding the utterance to the queue while the global SpeechSynthesis instance is in the paused state,
     * and then calling the resume method does not cause the resume event to be fired,
     * in this case the utterance’s start event will be called when the utterance starts.
     */
    onresume, 
    /**
     * Fired when the spoken utterance reaches a named "mark" tag in SSML.
     * [SSML] The user agent must fire this event if the speech synthesis engine provides the event.
     */
    onmark, 
    /**
     * Fired when the spoken utterance reaches a word or sentence boundary.
     * The user agent must fire this event if the speech synthesis engine provides the event.
     */
    onboundary) {
        this.internal = window.speechSynthesis;
        this._config = config;
        this._lang = lang;
        if (typeof voice === 'string') {
            this.internal
                .addEventListener('voiceschanged', () => {
                this._voice = this.internal
                    .getVoices().find(v => v.name === voice);
            });
        }
        else {
            this._voice = voice;
        }
        this._volume = volume;
        this._rate = rate;
        this._pitch = pitch;
        this._onstart = onstart;
        this._onend = onend;
        this._onerror = onerror;
        this._onpause = onpause;
        this._onresume = onresume;
        this._onmark = onmark;
        this._onboundary = onboundary;
    }
    /**
     * This attribute specifies the language of the speech synthesis for the utterance,
     * using a valid BCP 47 language tag.
     * [BCP47] If unset it remains unset for getting in script,
     * but will default to use the language of the html document root element and associated hierarchy.
     * This default value is computed and used when the input request opens a connection
     * to the recognition service.
     * @return {?}
     */
    get lang() {
        return this._lang || this._config.lang;
    }
    /**
     * @param {?} lang
     * @return {?}
     */
    set lang(lang) {
        this._lang = lang;
    }
    /**
     * This attribute specifies the speech synthesis voice that the web application wishes to use.
     * When a SpeechSynthesisUtterance object is created this attribute must be initialized to null.
     * If, at the time of the speak() method call,
     * this attribute has been set to one of the SpeechSynthesisVoice objects returned by getVoices(),
     * then the user agent must use that voice. If this attribute is unset or null at the time of the speak()
     * method call, then the user agent must use a user agent default voice.
     * The user agent default voice should support the current language (see lang) and
     * can be a local or remote speech service and can incorporate end user choices via interfaces
     * provided by the user agent such as browser configuration parameters.
     * @return {?}
     */
    get voice() {
        return this._voice || this._config.voice;
    }
    /**
     * @param {?} voice
     * @return {?}
     */
    set voice(voice) {
        this._voice = voice;
    }
    /**
     * This attribute specifies the speaking volume for the utterance.
     * It ranges between 0 and 1 inclusive, with 0 being the lowest volume and 1 the highest volume,
     * with a default of 1. If SSML is used, this value will be overridden by prosody tags in the markup.
     * @return {?}
     */
    get volume() {
        return this._volume || this._config.volume;
    }
    /**
     * @param {?} volume
     * @return {?}
     */
    set volume(volume) {
        this._volume = volume;
    }
    /**
     * This attribute specifies the speaking rate for the utterance.
     * It is relative to the default rate for this voice.
     * 1 is the default rate supported by the speech synthesis engine or specific voice
     * (which should correspond to a normal speaking rate).
     * 2 is twice as fast, and 0.5 is half as fast. Values below 0.1 or above 10 are strictly disallowed,
     * but speech synthesis engines or specific voices may constrain the minimum and maximum rates further,
     * for example, a particular voice may not actually speak faster than 3 times normal
     * even if you specify a value larger than 3.
     * If SSML is used, this value will be overridden by prosody tags in the markup.
     * @return {?}
     */
    get rate() {
        return this._rate || this._config.rate;
    }
    /**
     * @param {?} rate
     * @return {?}
     */
    set rate(rate) {
        this._rate = rate;
    }
    /**
     * This attribute specifies the speaking pitch for the utterance.
     * It ranges between 0 and 2 inclusive, with 0 being the lowest pitch and 2 the highest pitch.
     * 1 corresponds to the default pitch of the speech synthesis engine or specific voice.
     * Speech synthesis engines or voices may constrain the minimum and maximum rates further.
     * If SSML is used, this value will be overridden by prosody tags in the markup.
     * @return {?}
     */
    get pitch() {
        return this._pitch || this._config.pitch;
    }
    /**
     * @param {?} pitch
     * @return {?}
     */
    set pitch(pitch) {
        this._pitch = pitch;
    }
    /**
     * Fired when this utterance has begun to be spoken.
     * @return {?}
     */
    get onstart() {
        return this._onstart || this._config.onstart;
    }
    /**
     * @param {?} onstart
     * @return {?}
     */
    set onstart(onstart) {
        this._onstart = onstart;
    }
    /**
     * Fired when this utterance has completed being spoken. If this event fires,
     * the error event must not be fired for this utterance.
     * @return {?}
     */
    get onend() {
        return this._onend || this._config.onend;
    }
    /**
     * @param {?} onend
     * @return {?}
     */
    set onend(onend) {
        this._onend = onend;
    }
    /**
     * Fired if there was an error that prevented successful speaking of this utterance.
     * If this event fires, the end event must not be fired for this utterance.
     * @return {?}
     */
    get onerror() {
        return this._onerror || this._config.onerror;
    }
    /**
     * @param {?} onerror
     * @return {?}
     */
    set onerror(onerror) {
        this._onerror = onerror;
    }
    /**
     * Fired when and if this utterance is paused mid-utterance.
     * @return {?}
     */
    get onpause() {
        return this._onpause || this._config.onpause;
    }
    /**
     * @param {?} onpause
     * @return {?}
     */
    set onpause(onpause) {
        this._onpause = onpause;
    }
    /**
     * Fired when and if this utterance is resumed after being paused mid-utterance.
     * Adding the utterance to the queue while the global SpeechSynthesis instance is in the paused state,
     * and then calling the resume method does not cause the resume event to be fired,
     * in this case the utterance’s start event will be called when the utterance starts.
     * @return {?}
     */
    get onresume() {
        return this._onresume || this._config.onresume;
    }
    /**
     * @param {?} onresume
     * @return {?}
     */
    set onresume(onresume) {
        this._onresume = onresume;
    }
    /**
     * Fired when the spoken utterance reaches a named "mark" tag in SSML.
     * [SSML] The user agent must fire this event if the speech synthesis engine provides the event.
     * @return {?}
     */
    get onmark() {
        return this._onmark || this._config.onmark;
    }
    /**
     * @param {?} onmark
     * @return {?}
     */
    set onmark(onmark) {
        this._onmark = onmark;
    }
    /**
     * Fired when the spoken utterance reaches a word or sentence boundary.
     * The user agent must fire this event if the speech synthesis engine provides the event.
     * @return {?}
     */
    get onboundary() {
        return this._onboundary || this._config.onboundary;
    }
    /**
     * @param {?} onboundary
     * @return {?}
     */
    set onboundary(onboundary) {
        this._onboundary = onboundary;
    }
    /**
     * This attribute specifies the text to be synthesized and spoken for this utterance.
     * This may be either plain text or a complete, well-formed SSML document.
     * [SSML] For speech synthesis engines that do not support SSML,
     * or only support certain tags, the user agent or speech engine must strip away
     * the tags they do not support and speak the text. There may be a maximum length of the text,
     * it may be limited to 32,767 characters.
     * @param {?} text
     * @return {?}
     */
    text(text) {
        /** @type {?} */
        const utterance = new SpeechSynthesisUtterance(text);
        utterance.lang = this.lang;
        utterance.voice = this.voice;
        utterance.volume = this.volume;
        utterance.rate = this.rate;
        utterance.pitch = this.pitch;
        utterance.onstart = this.onstart;
        utterance.onend = this.onend;
        utterance.onerror = this.onerror;
        utterance.onpause = this.onpause;
        utterance.onresume = this.onresume;
        utterance.onmark = this.onmark;
        utterance.onboundary = this.onboundary;
        return utterance;
    }
}
SpeechSynthesisUtteranceFactoryService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
SpeechSynthesisUtteranceFactoryService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [Config,] }] },
    { type: String, decorators: [{ type: Optional }, { type: Inject, args: [Lang,] }] },
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [Voice,] }] },
    { type: Number, decorators: [{ type: Optional }, { type: Inject, args: [Volume,] }] },
    { type: Number, decorators: [{ type: Optional }, { type: Inject, args: [Rate,] }] },
    { type: Number, decorators: [{ type: Optional }, { type: Inject, args: [Pitch,] }] },
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [OnStartHandler,] }] },
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [OnEndHandler,] }] },
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [OnErrorHandler,] }] },
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [OnPauseHandler,] }] },
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [OnResumeHandler,] }] },
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [OnMarkHandler,] }] },
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [OnBoundaryHandler,] }] }
];
/** @nocollapse */ SpeechSynthesisUtteranceFactoryService.ngInjectableDef = i0.defineInjectable({ factory: function SpeechSynthesisUtteranceFactoryService_Factory() { return new SpeechSynthesisUtteranceFactoryService(i0.inject(i1.Config, 8), i0.inject(i1.Lang, 8), i0.inject(i1.Voice, 8), i0.inject(i1.Volume, 8), i0.inject(i1.Rate, 8), i0.inject(i1.Pitch, 8), i0.inject(i1.OnStartHandler, 8), i0.inject(i1.OnEndHandler, 8), i0.inject(i1.OnErrorHandler, 8), i0.inject(i1.OnPauseHandler, 8), i0.inject(i1.OnResumeHandler, 8), i0.inject(i1.OnMarkHandler, 8), i0.inject(i1.OnBoundaryHandler, 8)); }, token: SpeechSynthesisUtteranceFactoryService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    SpeechSynthesisUtteranceFactoryService.prototype._config;
    /**
     * @type {?}
     * @private
     */
    SpeechSynthesisUtteranceFactoryService.prototype._lang;
    /**
     * @type {?}
     * @private
     */
    SpeechSynthesisUtteranceFactoryService.prototype._voice;
    /**
     * @type {?}
     * @private
     */
    SpeechSynthesisUtteranceFactoryService.prototype._volume;
    /**
     * @type {?}
     * @private
     */
    SpeechSynthesisUtteranceFactoryService.prototype._rate;
    /**
     * @type {?}
     * @private
     */
    SpeechSynthesisUtteranceFactoryService.prototype._pitch;
    /**
     * @type {?}
     * @private
     */
    SpeechSynthesisUtteranceFactoryService.prototype._onstart;
    /**
     * @type {?}
     * @private
     */
    SpeechSynthesisUtteranceFactoryService.prototype._onend;
    /**
     * @type {?}
     * @private
     */
    SpeechSynthesisUtteranceFactoryService.prototype._onerror;
    /**
     * @type {?}
     * @private
     */
    SpeechSynthesisUtteranceFactoryService.prototype._onpause;
    /**
     * @type {?}
     * @private
     */
    SpeechSynthesisUtteranceFactoryService.prototype._onresume;
    /**
     * @type {?}
     * @private
     */
    SpeechSynthesisUtteranceFactoryService.prototype._onmark;
    /**
     * @type {?}
     * @private
     */
    SpeechSynthesisUtteranceFactoryService.prototype._onboundary;
    /**
     * If SpeechSynthesis API is supported
     * by the browser instance will be included.
     * @type {?}
     * @private
     */
    SpeechSynthesisUtteranceFactoryService.prototype.internal;
}
