// @kamiazya/ngx-speech-synthesis/esm5/lib/speech-synthesis.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { Config } from './configs';
var SpeechSynthesisModule = /** @class */ (function () {
    function SpeechSynthesisModule() {
    }
    /**
     * @param {?} config
     * @return {?}
     */
    SpeechSynthesisModule.forRoot = /**
     * @param {?} config
     * @return {?}
     */
    function (config) {
        return {
            ngModule: SpeechSynthesisModule,
            providers: [
                {
                    provide: Config,
                    useValue: config,
                },
            ],
        };
    };
    SpeechSynthesisModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [],
                    imports: [],
                    exports: []
                },] }
    ];
    return SpeechSynthesisModule;
}());
export { SpeechSynthesisModule };
