// @kamiazya/ngx-speech-synthesis/esm2015/lib/configs.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InjectionToken } from '@angular/core';
/**
 * @record
 */
export function SpeechSynthesisUtteranceParams() { }
if (false) {
    /** @type {?|undefined} */
    SpeechSynthesisUtteranceParams.prototype.lang;
    /** @type {?|undefined} */
    SpeechSynthesisUtteranceParams.prototype.pitch;
    /** @type {?|undefined} */
    SpeechSynthesisUtteranceParams.prototype.rate;
    /** @type {?|undefined} */
    SpeechSynthesisUtteranceParams.prototype.text;
    /** @type {?|undefined} */
    SpeechSynthesisUtteranceParams.prototype.voice;
    /** @type {?|undefined} */
    SpeechSynthesisUtteranceParams.prototype.volume;
}
/**
 * @record
 */
export function SpeechSynthesisUtteranceConfig() { }
if (false) {
    /** @type {?|undefined} */
    SpeechSynthesisUtteranceConfig.prototype.onboundary;
    /** @type {?|undefined} */
    SpeechSynthesisUtteranceConfig.prototype.onend;
    /** @type {?|undefined} */
    SpeechSynthesisUtteranceConfig.prototype.onerror;
    /** @type {?|undefined} */
    SpeechSynthesisUtteranceConfig.prototype.onmark;
    /** @type {?|undefined} */
    SpeechSynthesisUtteranceConfig.prototype.onpause;
    /** @type {?|undefined} */
    SpeechSynthesisUtteranceConfig.prototype.onresume;
    /** @type {?|undefined} */
    SpeechSynthesisUtteranceConfig.prototype.onstart;
}
/** @type {?} */
export const Lang = new InjectionToken('speech-synthesis.lang');
/** @type {?} */
export const Voice = new InjectionToken('speech-synthesis.voice');
/** @type {?} */
export const Volume = new InjectionToken('speech-synthesis.volume');
/** @type {?} */
export const Rate = new InjectionToken('speech-synthesis.rate');
/** @type {?} */
export const Pitch = new InjectionToken('speech-synthesis.pitch');
/** @type {?} */
export const OnStartHandler = new InjectionToken('speech-synthesis.onstart');
/** @type {?} */
export const OnEndHandler = new InjectionToken('speech-synthesis.onend');
/** @type {?} */
export const OnErrorHandler = new InjectionToken('speech-synthesis.onerror');
/** @type {?} */
export const OnPauseHandler = new InjectionToken('speech-synthesis.onpause');
/** @type {?} */
export const OnResumeHandler = new InjectionToken('speech-synthesis.onresume');
/** @type {?} */
export const OnMarkHandler = new InjectionToken('speech-synthesis.onmark');
/** @type {?} */
export const OnBoundaryHandler = new InjectionToken('speech-synthesis.onboundary');
/** @type {?} */
export const Params = new InjectionToken('speech-synthesis.params');
/** @type {?} */
export const Config = new InjectionToken('speech-synthesis.config');
